src/Sender/CreditMemoEmailSender.php

```diff
@@ -22,36 +22,34 @@ final class CreditMemoEmailSender implements CreditMemoEmailSenderInterface
 {
     private const UNITS_REFUNDED = 'units_refunded';
 
-    private CreditMemoPdfFileGeneratorInterface $creditMemoPdfFileGenerator;
-
-    private SenderInterface $sender;
-
-    private FileManagerInterface $fileManager;
-
     public function __construct(
-        CreditMemoPdfFileGeneratorInterface $creditMemoPdfFileGenerator,
-        SenderInterface $sender,
-        FileManagerInterface $fileManager
+        private CreditMemoPdfFileGeneratorInterface $creditMemoPdfFileGenerator,
+        private SenderInterface $sender,
+        private FileManagerInterface $fileManager,
+        private bool $hasEnabledPdfFileGenerator = true
     ) {
-        $this->creditMemoPdfFileGenerator = $creditMemoPdfFileGenerator;
-        $this->sender = $sender;
-        $this->fileManager = $fileManager;
     }
 
     public function send(CreditMemoInterface $creditMemo, string $recipient): void
     {
-        $creditMemoPdfFile = $this->creditMemoPdfFileGenerator->generate($creditMemo->getId());
+        if ($this->hasEnabledPdfFileGenerator) {
+            $creditMemoPdfFile = $this->creditMemoPdfFileGenerator->generate($creditMemo->getId());
+
+            $filePath = $creditMemoPdfFile->filename();
+            $this->fileManager->createWithContent($filePath, $creditMemoPdfFile->content());
 
-        $filePath = $creditMemoPdfFile->filename();
-        $this->fileManager->createWithContent($filePath, $creditMemoPdfFile->content());
+            $attachments = [$this->fileManager->realPath($filePath)];
+        }
 
         $this->sender->send(
             self::UNITS_REFUNDED,
             [$recipient],
             ['creditMemo' => $creditMemo],
-            [$this->fileManager->realPath($filePath)]
+            $attachments ?? []
         );
 
-        $this->fileManager->remove($filePath);
+        if ($this->hasEnabledPdfFileGenerator) {
+            $this->fileManager->remove($filePath ?? '');
```

Comment:
Workaround because of:
https://github.com/Sylius/RefundPlugin/runs/6120425251?check_suite_focus=true#step:28:19

Reply:
Should be needed after minor refactor according to the above comment ☝🏻